linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java

```diff
@@ -3263,7 +3267,10 @@ private boolean advance() {
         TKey leftKey2 = outerKeySelector.apply(left);
         int c = leftKey.compareTo(leftKey2);
         if (c != 0) {
-          assert c < 0 : "not sorted";
+          if (c < 0) {
+            throw new IllegalStateException(
+              "not sorted, " + leftKey + " is less than " + leftKey2);
```

Comment:
The message should be updated to account `c>0`.
A bit more verbose message won't hurt.

For instance: "mergeJoin assumes sorted in ascending order input, however ... "